web-module/src/main/webapp/templates/celTemplates/newsletterAddressList.vm

```diff
@@ -34,49 +43,64 @@
-	#foreach($emailAdr in $emailActiveList)
-	#if ("$!orderByDate" == "1")
-	#set($mailLang = '-')##
-	#if("$!{emailAdr.getLanguage()}" != '')##
-	#set($mailLang = "$!{emailAdr.getLanguage()}")##
-	#end##
-	$emailAdr.getEmailAdr()&#09;$mailLang&#09;$datetool.format('dd.MM.yyyy HH:mm', $emailAdr.getChangeDate())
-	#else
-	$emailAdr
-	#end
-	#end
-	</textarea>  
-	</div>
-	
-	<div style="float: left;padding-left:10px;">
-	<strong>Inaktive ($inactiveCount):</strong><br/>
-	<textarea name="inactive" id="textarea_inactive" cols="60" rows="30">
-	#foreach($mailDocName in $emailInactiveDocNameList)
-	#set($mailDoc = $xwiki.getDocument($mailDocName))
-	#set($mailObj = $mailDoc.getObject('Celements.NewsletterReceiverClass', 'subscribed', "${blogname}", false))
-	#set($emailAdr = "$!mailObj.getProperty('email').getValue()")
-	#if ("$!orderByDate" == "1")
-	#set($emailLang = "$!{mailObj.getProperty('language').getValue()}")
-	#if("$!emailLang" == '')
-	  #set($emailLang = "-")
-	#end
-	$emailAdr&#09;$emailLang&#09;$datetool.format('dd.MM.yyyy HH:mm', $mailDoc.date)
-	#else
-	$emailAdr
-	#end
-	#end
-	</textarea>
-	
-	</div>
+  <span style="font-size: 10px;"><a href='$doc.getURL("view","xpage=overlay&conf=NewsletterAddressList&xredirect=${request.xredirect}&orderByDate=$!{orderByDateParam}&namesVisible=$!{namesVisible}")'>$sortText</a>
+  </span>
+  <span style="font-size: 10px;"><a href='$doc.getURL("view","xpage=overlay&conf=NewsletterAddressList&xredirect=${request.xredirect}&orderByDate=$!{orderByDate}&namesVisible=$!{namesVisibleParam}")'>$namesVisibleText</a>
+  </span>
+  </h1>
+  <br/>
+  <div style="float: left;">
+  <strong>Aktive ($activeCount):</strong><br/>
+  #set($activeAddresses = "")
+  #foreach($emailAdr in $emailActiveList)
+    #if("$!orderByDate" == "1")
+      #set($mailLang = '-')##
+      #if("$!{emailAdr.getLanguage()}" != '')##
+        #set($mailLang = "$!{emailAdr.getLanguage()}")##
+      #end##
+      #set($activeAddresses = "$!{activeAddresses}$emailAdr.getEmailAdr()&#09;$mailLang&#09;$datetool.format('dd.MM.yyyy HH:mm', $emailAdr.getChangeDate())${xwiki.nl}")
+    #else
+      #set($activeAddresses = "$!{activeAddresses}$emailAdr${xwiki.nl}")
+    #end
+  #end ## foreach
+  <textarea name="active" id="textarea_active" cols="60" rows="30">$!{activeAddresses}</textarea>
+  </div>
+
+  <div style="float: left;padding-left:10px;">
+  <strong>Inaktive ($inactiveCount):</strong><br/>
+  #set($inactiveAddresses = "")
+  #foreach($mailDocName in $emailInactiveDocNameList)
+    #set($mailDoc = $xwiki.getDocument($mailDocName))
+    #set($mailObj = $mailDoc.getObject('Celements.NewsletterReceiverClass', 'subscribed', "${blogname}", false))
+    #set($contactObj = $mailDoc.getObject('Celements.ContactClass', false))
+    #set($emailAdr = "$!mailObj.getProperty('email').getValue()")
+    #set($conactName = "")
+    #if("$!namesVisible" == "1")
+      #set($firstname = "$!{contactObj.getProperty('firstname').getValue()}")

```

Comment:
Not necessarily. $!contactObj.firstname triggers Object.get("firstname") which itself ends in calling doc.display("firstname", ...). XWikiDocument.display depends on context.action and sometimes does some rendering. There is even a JavaDoc on Object.get mentioning the difference:
    /**
     * This method's name is misleading since it doesn't return the Object's property value; it"s equivalent to
     * {@link #display(String, String)} (with {@code type} equals to {@code view}). In order to get the Object's
     * property value use {@code getProperty(String).getValue()} instead.
     */